libarchive/archive_write_disk_posix.c

```diff
@@ -1648,6 +1638,17 @@ _archive_write_disk_finish_entry(struct archive *_a)
 	}
 
 	/*
+	 * Security-related extended attributes (such as
+	 * security.capability or security.ima on Linux) have to be restored last,
+	 * since they're implicitly removed by other file changes like setting
+	 * times.
+	 */
+	if (a->todo & TODO_XATTR) {
+		int r2 = set_xattrs(a);
+		if (r2 < ret) ret = r2;
+	}
+
+	/*
```

Comment:
Are you sure this must follow fflags setting?  In particular, setting schg should prevent subsequent setting of xattrs.